packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3782,55 +3784,69 @@ class _Editable extends MultiChildRenderObjectWidget {
   }
 }
 
+/// A function that that takes a placeholder location as an int offset into some
+/// text.
+typedef _PlaceholderLocationCallback = void Function(int location);
+
+/// A function that takes the Offset at which focus is requested.
+typedef _ScribbleFocusCallback = void Function(Offset offset);
+
+/// A widget that provides the ability to receive handwriting input from
+/// [Scribble].
 class _ScribbleFocusable extends StatefulWidget {
   const _ScribbleFocusable({
     required this.child,
-    required this.focusNode,
-    required this.editableKey,
-    required this.updateSelectionRects,
     required this.enabled,
+    required this.onPlaceholderLocationChanged,
+    required this.onScribbleFocus,
+    required this.onShowToolbar,
+    required this.readOnly,
+    required this.value,
   });
 
   final Widget child;
-  final FocusNode focusNode;
-  final GlobalKey editableKey;
-  final VoidCallback updateSelectionRects;
   final bool enabled;
+  final _PlaceholderLocationCallback onPlaceholderLocationChanged;
+  final _ScribbleFocusCallback onScribbleFocus;
+  final VoidCallback onShowToolbar;
+  final bool readOnly;
+  final TextEditingValue value;
 
   @override
   _ScribbleFocusableState createState() => _ScribbleFocusableState();
 }
 
-class _ScribbleFocusableState extends State<_ScribbleFocusable> implements ScribbleClient {
+class _ScribbleFocusableState extends State<_ScribbleFocusable> with ScribbleClient {
   _ScribbleFocusableState(): _elementIdentifier = (_nextElementIdentifier++).toString();
 
   @override
   void initState() {
     super.initState();
+    Scribble.client = this;
```

Comment:
Is this right? The documentation states:
 
>  /// Set the given [ScribbleClient] as the single active client.
>  ///
>  /// This is usually based on the [ScribbleClient] receiving focus.

Reply:
Good call, the docs are right, it should be set on focus.  I'll fix that and add a test.